File: lokasiku-backend/api-service/src/main/java/org/lokasiku/apiservice/config/PhysicalNamingStrategy.java
package org.lokasiku.apiservice.config;

import org.hibernate.boot.model.naming.Identifier;
import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy;

public class PhysicalNamingStrategy extends SpringPhysicalNamingStrategy {

  @Override
  public Identifier toPhysicalTableName(Identifier name, JdbcEnvironment jdbcEnvironment) {
    // Warning. Hacky code.
    var isClassName = Character.isUpperCase(name.getText().charAt(0));
    var newName = isClassName ? super.toPhysicalTableName(name, jdbcEnvironment).getText() + "s" : name.getText();
    return Identifier.toIdentifier(newName);
  }

}

File: lokasiku-backend/api-service/src/main/kotlin/org/lokasiku/apiservice/config/PhysicalNamingStrategy.kt
package org.lokasiku.apiservice.config

import org.hibernate.boot.model.naming.Identifier
import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment
import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy

class PhysicalNamingStrategy : SpringPhysicalNamingStrategy() {

    override fun toPhysicalTableName(name: Identifier?, jdbcEnvironment: JdbcEnvironment?): Identifier {
        // Warning. Hacky implementation to determine table name.
        val newName = when(name?.text?.get(0)?.isUpperCase()) {
            true ->  "${super.toPhysicalTableName(name, jdbcEnvironment).text}s"
            false, null -> name?.text
        }
        return Identifier.toIdentifier(newName)
    }
}